packages/amplify_storage_s3/ios/Classes/AmplifyStorageOperations.swift

```diff
@@ -141,16 +144,59 @@ public class AmplifyStorageOperations {
         return itemAsDictionary
     }
     
-    private static func prepareError(flutterResult: @escaping FlutterResult, msg: String, localizedError: String?, recoverySuggestion: String?) {
-        var errorDetails: [String: Any] = [:]
-        var platformErrorDetails: [String: String] = [:]
-        
-        platformErrorDetails["platform"] = "iOS"
-        platformErrorDetails["localalizedError"] = localizedError
-        platformErrorDetails["recoverySuggestion"] = recoverySuggestion
-        
-        errorDetails["PLATFORM_EXCEPTIONS"] = platformErrorDetails
-        
-        flutterResult(FlutterError(code: "AmplifyException", message: msg, details: errorDetails))
+    private static func prepareError(flutterResult: @escaping FlutterResult, storageError: StorageError) {
+        let details: Dictionary<String, String> = createSerializedError(storageError: storageError);
+        logErrorContents(error: storageError)
+           ErrorUtil.postErrorToFlutterChannel(
+               result: flutterResult,
+               errorCode: "StorageException",
+               details: details)
+    }
+    
+    private static func prepareGenericError(flutterResult: @escaping FlutterResult, error: Error) {
+        logErrorContents(error: error)
+        var message: String = ErrorMessages.missingExceptionMessage
+        var recoverySuggestion: String = ErrorMessages.missingRecoverySuggestion
```

Comment:
`ErrorMessages.missingExceptionMessage` is not really accurate.

In Android you have this as `ExceptionMessages.unexpectedExceptionMessage`. We should try to make it consistent. Are the messages the same btw?

Reply:
This is being used by all other packages.